src/goto-cc/compile.cpp

```diff
@@ -421,9 +421,10 @@ bool compilet::compile()
       std::string cfn;
 
       if(output_file_object=="")
-        cfn=get_base_name(file_name, true)+"."+object_file_extension;
+        cfn = output_directory_object + get_base_name(file_name, true) + "." +
```

Comment:
I believe this lacks a path separator after `output_directory_object` (which should be appended to `output_directory_object` if it is non-empty).

Reply:
done